src/base/SolutionArray.cpp

```diff
@@ -1273,6 +1277,23 @@ void SolutionArray::writeEntry(AnyMap& root, const string& name, const string& s
         data["components"] = componentNames();
     }
 
+    // add ordering rules
+    vector<vector<string>> ordering = {};
+    if (data.hasKey("components")) {
+        auto components = data["components"].as<vector<string>>();
+        for (auto component : boost::adaptors::reverse(components)) {
+            ordering.push_back({"head", component});
+        }
+    }
+
+    const vector<string> header = { "type", "size", "basis", "components" };
+    for (auto entry : boost::adaptors::reverse(header)) {
+        ordering.push_back({"head", entry});
+    }
+
+    data["__type__"] = "SolutionArray";
+    AnyMap::addOrderingRules("SolutionArray", ordering);
```

Comment:
This is problematic -- the `addOrderingRules` method is meant to be called exactly once for a given "type", hence why it's a static method. This usage will result in a memory leak as duplicate fields will be added to the ordering rules every time this method is called.

 The mechanism we currently have for setting order for a dynamic set of keys is through the `setLoc` method, though that may be an imperfect fit here.

Reply:
Sorry Ray! Thanks for catching this

Reply:
@speth ... likewise thanks for catching. Hadn't used that feature before - will create an issue.